strategy/to_json/from_serializable_test.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018-2021 Gamebridge.ai authors. All rights reserved. MIT license.
+
+import { assertEquals, test } from "../../test_deps.ts";
+import { Serializable } from "../../serializable.ts";
+import {
+  SerializeProperty,
+  SerializePropertyArgument,
+} from "../../serialize_property.ts";
+import { fromSerializable } from "./from_serializable.ts";
+
+function FromSerializable(
+  propertyName?: string,
+): PropertyDecorator {
+  const opts: SerializePropertyArgument = {
+    toJSONStrategy: fromSerializable,
+  };
+  if (propertyName) {
+    opts.serializedKey = propertyName;
+  }
+  return SerializeProperty(opts);
+}
+
+test({
+  name: "fromSerializable - arrays of nested objects",
+  fn() {
+    class Test1 extends Serializable {
+      @SerializeProperty("nested_property")
+      serializeMe = 999;
+    }
+    class Test2 extends Serializable {
+      @FromSerializable("outer_property")
+      nested: Test1[] = [new Test1()];
+    }
+
+    class Test3 extends Serializable {
+      @FromSerializable("outer_outer_property")
+      nested2: Test2[] = [new Test2()];
```

Comment:
Is the `@FromSerializable` needed here? I would think if we had
```
@SerializeProperty("outer_outer_property")
nested2: Test2[] = [new Test2()];
```
... then the default strategy would map through the array and call tsSerialize() if it's serializable.

Reply:
ah so this change runs against the whole array now and not per item, because under the hood `toJSON()` is called and returns the string, so we need to manually say to use `tsSerialize`. This is actually done on purpose because JSON is a text format, and we are trying hard to follow JS API's. for example 

```ts
new Date().toJSON() // returns string value and not a POJO
```